packages/backend/src/helpers/flow-templates.ts

```diff
@@ -72,6 +145,35 @@ export async function createFlowFromTemplate(
     })
 
     validateAppAndEventKey(steps[0], flowName)
+    // create a tile with its columns and rows first if template uses tiles
+    let tableId: string | undefined
+    const columnNameToIdMap: Record<string, string> = {}
+    if (tileTemplateData) {
+      const { name, columns, rowData } = tileTemplateData
+      const table = await user.$relatedQuery('tables').insertGraph({
+        name,
+        role: 'owner',
+        columns,
+      })
+      tableId = table.id
+
+      // obtain a map of column name to the column id for easy replacement
+      for (const tableColumn of table.columns) {
+        columnNameToIdMap[tableColumn.name] = tableColumn.id
+      }
+
+      // replace column names with column ids for each row data and create table rows
+      const newRowData = replaceRowData(rowData, columnNameToIdMap)
+      await createTableRows({ tableId, dataArray: newRowData })
+    }
+
+    // prepare placeholder map for replacement
+    const placeholderReplacementMap: IJSONObject = {}
+    placeholderReplacementMap[USER_EMAIL_PLACEHOLDER] = user.email
+    placeholderReplacementMap[TILE_ID_PLACEHOLDER] = tableId
+    placeholderReplacementMap[TILE_COL_DATA_PLACEHOLDER] = columnNameToIdMap
```

Comment:
nit: not sure why u dont wanna just declare the object directly 🤔 
e.g. 
const placeholderMap = {
   [USER_EMAIL_PLACEHOLDER]: user.email
   ...
}